### PR TITLE
[Upmeter] Update internal dev doc

### PR DESCRIPTION
**Pull request by shvgn**
## Description

Update internal doc, add diagrams for control-plane probe s

## Changelog entries

```changes
section: upmeter
type: chore
summary: Updated internal doc
```

### CODE REVIEW

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 1]
```diff
@@ -2,29 +2,89 @@
 title: "Информация для разработчиков"
```
Comment (shvgn) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:2]: I'd call it CONCEPTS, not 'development'.

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 2]
```diff
@@ -2,29 +2,89 @@
...
 
-- Хранит всё в базе данных SQLite, в файле `/db/dowtime.db.sqlite`.
+## Агент
 
-- Умеет принимать данные от агентов:
-  - Сразу дедуплицирует данные из 30-секундных таймслотов в 5-минутные (выбирая “лучший” результат);
-  - Складывает данные в SQLite.
+Агент измеряет доступность компонентов Deckhouse. Измерение состоит в проверке состояния объекта в
+апи кластера или проверке делает ответа на HTTP-запрос к приложению. Например, что состояние пода
+Ready или что Prometheus отвечает корректно на HTTP-запрос.
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:28]:
```suggestion
Агент измеряет доступность компонентов Deckhouse. Измерение состоит в проверке состояния объекта через API кластера или проверке ответа на HTTP-запрос к приложению. Например, что состояние Pod'а
Ready или что Prometheus отвечает корректно на HTTP-запрос.
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 3]
```diff
@@ -2,29 +2,89 @@
...
 
-- Хранит 30-секундные таймслоты только за сутки, старое удаляет из таблицы.
-- Хранит 5-минутные таймслоты постоянно, ничего не удаляет.
+Логическая единица доступности называется пробой. Проба тестирует некоторую функциональность. Проба
+состоит из одной или нескольких параллельных проверок. Например, проба `cluster-scaling` состоит из
+трех проверок, проверяющих статус подов `cloud-controller-manager`, `machine-controller-manager` и
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:32]:
```suggestion
трех проверок, проверяющих статус Pod'ов `cloud-controller-manager`, `machine-controller-manager` и
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 4]
```diff
@@ -2,29 +2,89 @@
...
+`bashible-apiserver`.
 
-- Умеет читать CRD `Downtime`, в этих CRD:
+Если одна из проверок выявила недоступность, компонента, статус пробы принимается `down`. Помимо
+недоступного, статус может быть `up`, если нет недоступных результатов проверок, или `uncertain`,
+если все проверки не смогли установить доступность или недоступность компонентов. Так может
+получиться, если не выполнены условия, заложенные в проверку. Например, проверки, опирающиеся на
+статус пода, будут в статусе `uncertain`, если перед тем, как проверить под, не подтвердится
+доступность аписервера.
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:40]:
```suggestion
Если одна из проверок выявила недоступность, компонента, статус пробы принимается `down`. Помимо статуса `down`, статус может быть:  `up`, если нет недоступных результатов проверок; `uncertain`,
если все проверки не смогли установить доступность или недоступность компонентов. Так может
получиться, если не выполнены условия, заложенные в проверку. Например, проверки, опирающиеся на
статус Pod'а, будут в статусе `uncertain`, если перед тем как проверить Pod не подтвердится
доступность API-сервера.
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 5]
```diff
@@ -2,29 +2,89 @@
...
+
+- длительность доступности (uptime)
+- длительность недоступности (downtime)
+- длительность неопределенности (uncertain)
+- оставшееся время из 30 секунд, за которое изменения не проводилось (nodata).
+
+Измерение не проводится только в том случае, если агент не запущен.
+
+Пробы объединяются в группы доступности. На эти группы выдается SLA. Статистика доступности группы
+так же вычисляется агентом во время сбора статистики проб. Статус группы вычисляется из статусов
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:59]:
```suggestion
также вычисляется агентом во время сбора статистики проб. Статус группы вычисляется из статусов
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 6]
```diff
@@ -2,29 +2,89 @@
...
+проб так же, как статус пробы вычисляется из статусов проверок.
+
+Статистика доступности проб и групп за 30 секунд отправляется в `upmeter server` по HTTP.
+
+Agent — это DaemonSet, который запускается только на узлах control plane. Поды агента используют
+SQLite для «WAL». Поэтому если `upmeter` недоступен, данные будут отправлены, когда тот поднимется.
+Данные в WAL ограничены последними 24 часами.
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:66]:
```suggestion
Agent — это DaemonSet, который запускается только на узлах control plane. Pod'ы агента используют
SQLite для «WAL». Поэтому если `upmeter` недоступен, данные будут отправлены, когда тот поднимется.
Данные в WAL ограничены последними 24 часами.
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md (hunk 5 above)
Comment (shvgn) [reply modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:59]: it's not "too", it is "similarly" :)

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 7]
```diff
@@ -2,29 +2,89 @@
...
+
+## Сервер
+
+Принимает статистику доступности за 30 секунд от агентов и складывает в статистику за 5 минут. Если
+в кластере более одного агента, сервер выбирает лучший вариант статистики. 5-минутная статистика
+хранится за все доступное время. БД для хранения — SQLite, файл `/db/dowtime.db.sqlite`.
+
+Сервер отдает данные в виде JSON. Этими данным пользуются дашборд апметра и статус-страница.
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:74]:
```suggestion
Сервер отдает данные в виде JSON. Этими данным пользуются дашборд upmeter и статус-страница.
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 8]
```diff
@@ -45,64 +105,127 @@ metadata:
...
-- monitoring-and-autoscaling
-- extensions-availability
 
-## smoke-mini
+### Control Plane
+
+#### Жизненный цикл объекта
+
+Проба направлена на доступность аписервера. Созданием и удалением объекта проверяется простой
+жизненный цикл объекта. Ошибки операций с аписервером засчитываются как нерабочее состояние.
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:185]:
```suggestion
Проба направлена на доступность API-сервера. Созданием и удалением объекта проверяется простой
жизненный цикл объекта. Ошибки операций с API-сервером засчитываются как нерабочее состояние.
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 9]
```diff
@@ -45,64 +105,127 @@ metadata:
...
+- `Basic Functionality`, создается и удаляется ConfigMap
+- `Namespace`, создается и удаляется Namespace
+
+![](01-single-object-lifecycle.png)
+
+#### Жизненный цикл состояния объекта
+
+Проба направлена на определение состояние объекта в его жизненном цикле. Проба считается
+непрошедшей, если объект не приобретает ожидаемого состояния. Например, если Pod не шедулится на
+ноду. Ошибки операций с аписервером считаются условием для неопределенного результата пробы.
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:198]:
```suggestion
узел. Ошибки операций с аписервером считаются условием для неопределенного результата пробы.
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 10]
```diff
@@ -45,64 +105,127 @@ metadata:
...
+- `Cert Manager`: Certificate → Secret.
+
+![](03-parent-child-lifecycle.png)
+
+### Synthetic
 
-Пробы из группы "synthetic" опрашивают smoke-mini — приложение, имитирующее настоящее. Это позволяет оценить, как будут вести себя настоящие приложения в кластере. `smoke-mini` запускает три `StatefulSet`, использующих `PV` и каждый имеющий 1 реплику, со специальным приложением, поднимающим HTTP-сервер и предоставляющим API для выполнения тестов. Ресурсы одного из `StatefulSet` перешедуливаются раз в 10 минут на случайные узлы.
+Пробы из группы «synthetic» опрашивают smoke-mini — модельное приложение, веб-сервер. Это позволяет
+оценить, как будут вести себя настоящие приложения в кластере. `smoke-mini` запускает пять
+StatefulSet с HTTP-сервером и предоставляющим API для выполнения тестов. Под одного из StatefulSet
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:222]:
```suggestion
StatefulSet с HTTP-сервером и предоставляющим API для выполнения тестов. Pod одного из StatefulSet
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 11]
```diff
@@ -45,64 +105,127 @@ metadata:
...
+#### Функционал тестирования
 
-* `/` – return 200;
-* `/error` – return 500;
-* `/api` – проверяет доступ к API Kubernetes (запрашивается информация по Pod'у, из которого выполняется запрос `/api/v1/namespaces/d8-smoke-mini/pods/<POD_NAME>`);
-* `/dns` – проверяет работу кластерного dns (выполняет резолв домена `kubernetes.default`);
-* `/disk` – проверяет, что может создать и удалить файл;
-* `/neighbor` – проверяет, есть ли доступ к "соседу" по HTTP;
-* `/prometheus` – проверяет, что может отправить запрос в Prometheus `/api/v1/metadata?metric=prometheus_build_info`.
+* `/` – доступность самого пода, всегда возращает код ответа 200
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:228]:
```suggestion
* `/` – доступность самого Pod'а, всегда возращает код ответа 200
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 12]
```diff
@@ -45,64 +105,127 @@ metadata:
...
+* `/dns` – проверяет работу кластерного DNS, выполняя резолв домена `kubernetes.default`
+* `/neighbor` – проверяет, есть ли доступ к «соседнему» StatefulSet по HTTP адресу пода
+* `/neighbor-via-service` – проверяет, есть ли доступ к «соседнему» StatefulSet через общий сервис
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:231]:
```suggestion
* `/` – доступность самого Pod'а, всегда возращает код ответа 200;
* `/dns` – проверяет работу кластерного DNS, выполняя разрешение доменого имени `kubernetes.default`;
* `/neighbor` – проверяет, есть ли доступ к «соседнему» StatefulSet по HTTP-адресу Pod'а;
* `/neighbor-via-service` – проверяет, есть ли доступ к «соседнему» StatefulSet через общий сервис.
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md (hunk 5 above)
Comment (z9r5) [reply modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:57]: Let's clarify. @shvgn Is it correct now?

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md [hunk 13]
```diff
@@ -2,29 +2,87 @@
 title: "Информация для разработчиков"
```
Comment (z9r5) [inline modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:2]:
```suggestion
title: "Описание архитектуры"
```

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md (hunk 1 above)
Comment (z9r5) [reply modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md]: OK. Made the changes.

File: modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md (hunk 5 above)
Comment (shvgn) [reply modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md:57]: Yup

**Review by z9r5 — APPROVED** [review]

**Pull request merged**